pra-07/src/views/carDetails.vue

```diff
@@ -0,0 +1,151 @@
+<template>
+  <div>
+    <div>
+      <AlertBox
+        v-if="this.isLoading == false && this.successmsg !== ''"
+        variant="success"
+        show="true"
+        :content="this.successmsg"
+      />
+      <AlertBox
+        v-if="this.isLoading === false && this.errmsg !== ''"
+        showAlert="true"
+        alertVariant="danger"
+        :alertMessage="this.errmsg"
+      />
+    </div>
+    <div class="d-flex justify-content- mt-5">
+      <transition name="bounce">
+        <FormModal
+          :formData="selectedCardData"
+          @submittedFormData="handleSubmittedData"
+        />
+      </transition>
+      <b-card
+        :title="this.$store.state.cars.carData.name"
+        :img-src="this.$store.state.cars.carData.image"
+        img-alt="Image"
+        img-top
+        tag="article"
+        style="max-width: 30rem"
+        class="mb-2"
+      >
+        <b-card-text>
+          {{ this.$store.state.cars.carData.details }}
+        </b-card-text>
+        <b-card-text>
+          price : {{ this.$store.state.cars.carData.price }}
+        </b-card-text>
+        <b-button
+          to="/"
+          pill
+          style="background-color: #1f2a48"
+          variant="outline-light"
+          >Back to homepage</b-button
+        >
+        <b-button
+          @click="deleteCard"
+          pill
+          style="background-color: #1f2a48"
+          variant="outline-light"
+          >delete</b-button
+        >
+        <b-button
+          @click="editCard"
+          pill
+          style="background-color: #1f2a48"
+          variant="outline-light"
+          >edit</b-button
+        >
+      </b-card>
+    </div>
+  </div>
+</template>
+
+<script>
+import AlertBox from "../components/AlertBox.vue";
+import FormModal from "../components/Form-Modal.vue";
+export default {
+  name: "CarDetailsPage",
+  components: { AlertBox, FormModal },
+
+  data() {
+    return {
+      cars: [],
+      formModalId: this.modalId,
+      selectedCardData: {
+        carId: "",
+        carName: "",
+        carDetails: "",
+        carPrice: "",
+        carImgURL: "",
+      },
+      submittedNames: [],
+      carData: {
+        image: "",
+        name: "",
+        details: "",
+        price: "",
+      },
+      isLoading: false,
+      errmsg: "",
+      successmsg: "",
+    };
+  },
+  methods: {
+    async getCarData(id) {
+      this.isLoading = true;
+      let res = await this.$store.dispatch("cars/getCarData", id);
+      if (res === "success") {
+        this.isLoading = false;
+      } else {
+        this.isLoading = false;
+        this.errmsg = "oops! somthing went wrong";
```

Comment:
in this.errormsg you can directly assign res because you have return error message in your getCarData action.